docs/10.demo.md

```diff
@@ -0,0 +1,169 @@
+## Demo Overview
+
+The WordPress Feature API Demo is a fully functional example plugin that demonstrates how to implement, register, and use the WordPress Features API in a real-world context. It showcases the API's capabilities through a practical implementation that includes both server-side and client-side features, a conversational interface, and tooling that demonstrates the extensibility of the system.
```

Comment:
I will merge this PR and then update the demo doc since a lot of this has changed in the rewrite.